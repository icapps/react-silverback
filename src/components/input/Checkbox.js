import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = props => (
  <div className="form-group">
    <div className="form-check">
      <input type="checkbox" className="form-check-input" id={props.id} checked={props.value} onChange={props.handleChange} disabled={props.isDisabled} />
      <label className="form-check-label" htmlFor={props.id}>{props.text}</label>
    </div>
  </div>
);

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  value: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool,
};

Checkbox.defaultProps = {
  isDisabled: false,
};

export default Checkbox;
