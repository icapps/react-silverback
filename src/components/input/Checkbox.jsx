import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = props => (
  <div className="form-check">
    <input type="checkbox" className="form-check-input" id={props.id} checked={props.value} onChange={props.handleChange} />
    <label className="form-check-label" htmlFor={props.id}>{props.text}</label>
  </div>
);

Checkbox.propTypes = {
  id: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  value: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default Checkbox;
