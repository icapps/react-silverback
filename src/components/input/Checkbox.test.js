import React from 'react';
import { shallow, configure } from 'enzyme';
import Checkbox from './Checkbox';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('BasicInput Component', () => {
  it('should render a BasicInput component', () => {
    const wrapper = shallow(<Checkbox id="test" text="This is a checkbox" value={true} handleChange={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
