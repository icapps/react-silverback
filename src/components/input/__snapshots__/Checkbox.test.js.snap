// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInput Component should render a BasicInput component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    css=""
    handleChange={[Function]}
    id="test"
    isDisabled={false}
    text="This is a checkbox"
    value={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="form-check"
>
        <input
                checked={true}
                className="form-check-input"
                disabled={false}
                id="test"
                onChange={[Function]}
                type="checkbox"
        />
        <label
                className="form-check-label"
                htmlFor="test"
        >
                This is a checkbox
        </label>
</div>,
      "className": "form-group ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={true}
            className="form-check-input"
            disabled={false}
            id="test"
            onChange={[Function]}
            type="checkbox"
/>,
          <label
            className="form-check-label"
            htmlFor="test"
>
            This is a checkbox
</label>,
        ],
        "className": "form-check",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "className": "form-check-input",
            "disabled": false,
            "id": "test",
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is a checkbox",
            "className": "form-check-label",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "This is a checkbox",
          "type": "label",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="form-check"
>
          <input
                    checked={true}
                    className="form-check-input"
                    disabled={false}
                    id="test"
                    onChange={[Function]}
                    type="checkbox"
          />
          <label
                    className="form-check-label"
                    htmlFor="test"
          >
                    This is a checkbox
          </label>
</div>,
        "className": "form-group ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={true}
              className="form-check-input"
              disabled={false}
              id="test"
              onChange={[Function]}
              type="checkbox"
/>,
            <label
              className="form-check-label"
              htmlFor="test"
>
              This is a checkbox
</label>,
          ],
          "className": "form-check",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": true,
              "className": "form-check-input",
              "disabled": false,
              "id": "test",
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is a checkbox",
              "className": "form-check-label",
              "htmlFor": "test",
            },
            "ref": null,
            "rendered": "This is a checkbox",
            "type": "label",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
